samcli/lib/cli_validation/image_repository_validation.py

```diff
@@ -85,7 +85,8 @@ def wrapped(*args, **kwargs):
                 Validator(
                     validation_function=lambda: not guided
                     and not (image_repository or image_repositories or resolve_image_repos)
-                    and required,
+                    and required
+                    and not _is_all_image_funcs_provided(template_file, image_repositories, parameters_overrides),
```

Comment:
Could we add a quick unit test just to confirm that if all the conditions are met (not guided, no image repo options passed, not required, no images set in template) that we indeed get that exception?